src/storage/src/sst.rs

```diff
@@ -285,19 +295,69 @@ impl FileHandleInner {
     }
 }
 
+#[derive(Debug, Snafu, PartialEq)]
+pub struct ParseIdError {
+    source: uuid::Error,
+}
+
+/// Unique id for [SST File].
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct FileId(Uuid);
+
+impl FileId {
+    /// Returns a new unique [FileId] randomly.
+    pub fn random() -> FileId {
+        FileId(Uuid::new_v4())
+    }
+
+    /// Parses id from string.
+    pub fn parse_str(input: &str) -> std::result::Result<FileId, ParseIdError> {
+        Uuid::parse_str(input).map(FileId).context(ParseIdSnafu)
+    }
+
+    /// Append `.parquet` to file id to make a complete file name
+    pub fn as_parquet(&self) -> String {
+        format!("{}{}", self.0.hyphenated(), ".parquet")
+    }
+}
+
+impl fmt::Display for FileId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl FromStr for FileId {
+    type Err = ParseIdError;
+
+    fn from_str(s: &str) -> std::result::Result<FileId, ParseIdError> {
+        FileId::parse_str(s)
+    }
+}
+
 /// Immutable metadata of a sst file.
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub struct FileMeta {
     /// Region of file.
     pub region_id: RegionId,
-    /// File name
-    pub file_name: String,
+    /// Compared to normal file names, FileId ignore the extension
+    #[serde(deserialize_with = "deserialize_from_string")]
+    pub file_name: FileId,
```

Comment:
We already customised the deserializer for `file_name`, why not change it to `file_id`?

Reply:
There are existed manifests that were serialized with `file_name` field. If we changed to `file_id`, these files can not be deserialized again. The deserialization is not tolerant with schema change, I think.

Reply:
Yep, I know that. But we can assume that the current users always upgrade their greptimedb instances not downgrade at all.  So the old instances can't read the new schema, it's not a big problem at this stage.

We can use `alias` to define multi alias of `file_id`

https://serde.rs/field-attrs.html#alias

I don't test it before, maybe `serde(alias = ["file_name", "file_id"])`.

Reply:
Looks like as below:
```rust
#serde(alias ="file_id")
#serde(alias ="file_name")
file_id: FileId,
```

Reply:
Make sense. I'll try.